From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Sat, 3 Jan 2009 18:25:03 +0100
Subject: [PATCH] use xulrunner not mozilla

---
 configure.ac |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 2131706..78b554b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -30,7 +30,7 @@ PYTHON_REQUIRED=2.4
 
 NSPR_REQUIRED=4.0.0
 FIREFOX_PLUGIN_REQUIRED=2.0.0
-MOZILLA_PLUGIN_REQUIRED=1.8
+MOZILLA_PLUGIN_REQUIRED=1.0
 
 dnl *******************************************************************************
 dnl Setup GNULIB - must be before anything else in this file
@@ -266,7 +266,7 @@ if test "x$enable_plugin" = "xyes"; then
     PKG_CHECK_MODULES(FIREFOX_PLUGIN,
                      firefox-plugin >= $FIREFOX_PLUGIN_REQUIRED, ,[
         PKG_CHECK_MODULES(MOZILLA_PLUGIN,
-                          mozilla-plugin >= $MOZILLA_PLUGIN_REQUIRED)
+                          iceape-plugin >= $MOZILLA_PLUGIN_REQUIRED)
         FIREFOX_PLUGIN_CFLAGS="$MOZILLA_PLUGIN_CFLAGS"
         FIREFOX_PLUGIN_LIBS="$MOZILLA_PLUGIN_LIBS"
     ])
-- 
