From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Fri, 23 Oct 2009 21:29:45 +0200
Subject: [PATCH] remove WERROR_CFLAGS

to fix build
---
 plugin/Makefile.am |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/plugin/Makefile.am b/plugin/Makefile.am
index 2fa8777..1bd70c3 100644
--- a/plugin/Makefile.am
+++ b/plugin/Makefile.am
@@ -12,7 +12,7 @@ gtk_vnc_plugin_la_LDFLAGS = \
 	-module -avoid-version
 gtk_vnc_plugin_la_CFLAGS = \
 	-I$(top_srcdir)/src @GTK_CFLAGS@ \
-	@FIREFOX_PLUGIN_CFLAGS@ @WARNING_CFLAGS@ @WERROR_CFLAGS@
+	@FIREFOX_PLUGIN_CFLAGS@ @WARNING_CFLAGS@
 
 all-local: gtk-vnc-plugin.so
 
-- 
