From: Michal Suchanek <hramrach@centrum.cz>
Date: Tue, 18 Aug 2009 18:23:00 +0200
Subject: [PATCH] allow connecting to local server by display number

Closes: #540434
---
 examples/gvncviewer.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/examples/gvncviewer.c b/examples/gvncviewer.c
index 777b8c8..b2cc30c 100644
--- a/examples/gvncviewer.c
+++ b/examples/gvncviewer.c
@@ -36,7 +36,7 @@ static const GOptionEntry options [] =
 {
   {
     G_OPTION_REMAINING, '\0', 0, G_OPTION_ARG_STRING_ARRAY, &args,
-    NULL, "hostname[:display]" },
+    NULL, "[hostname][:display]" },
   { NULL, 0, 0, G_OPTION_ARG_NONE, NULL, NULL, 0 }
 };
 
@@ -383,7 +383,7 @@ int main(int argc, char **argv)
 		return 1;
 	}
 	if (!args || (g_strv_length(args) != 1)) {
-		fprintf(stderr, "Usage: gvncviewer hostname[:display]\n%s\n", help_msg);
+		fprintf(stderr, "Usage: gvncviewer [hostname][:display]\n%s\n", help_msg);
 		return 1;
 	}
 
@@ -461,6 +461,8 @@ int main(int argc, char **argv)
 	} else
 		snprintf(port, sizeof(port), "%d", 5900);
 
+	if(! *hostname)
+		snprintf(hostname, sizeof(hostname), "%s", "127.0.0.1");
 	vnc_display_open_host(VNC_DISPLAY(vnc), hostname, port);
 	vnc_display_set_keyboard_grab(VNC_DISPLAY(vnc), TRUE);
 	vnc_display_set_pointer_grab(VNC_DISPLAY(vnc), TRUE);
-- 
