From: Joao Eriberto Mota Filho <eriberto@debian.org>
Date: Fri, 10 Feb 2017 10:22:10 +0100
Subject: Add -I m4 to Makefile.am

---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 69d1f50..dabb899 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,5 +1,5 @@
 SUBDIRS = src tools examples po vapi
-ACLOCAL_AMFLAGS = ${ACLOCAL_FLAGS}
+ACLOCAL_AMFLAGS = ${ACLOCAL_FLAGS} -I m4
 
 pkgconfig_DATA = $(PACKAGE)-$(GTK_VNC_API_VERSION).pc gvnc-1.0.pc
 pkgconfigdir = $(libdir)/pkgconfig
