diff --git a/plugin/Makefile.am b/plugin/Makefile.am
index 813e879..58ed449 100644
--- a/plugin/Makefile.am
+++ b/plugin/Makefile.am
@@ -21,7 +21,7 @@ gtk-vnc-plugin.so: gtk-vnc-plugin.la
 
 # Only leave the .so file in the plugins directory.
 install-data-hook:
-	rm -f $(plugindir)/gtk-vnc-plugin.a $(plugindir)/gtk-vnc-plugin.la
+	rm -f $(DESTDIR)$(plugindir)/gtk-vnc-plugin.a $(DESTDIR)$(plugindir)/gtk-vnc-plugin.la
 
 EXTRA_DIST = README
 
